### PR TITLE
msvc/arm: Fix for https://github.com/boostorg/type_traits/issues/105

### PR DESCRIPTION
**Pull request by jzmaddock**

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -41,6 +41,8 @@ environment:
       ARGS: --toolset=msvc-14.1 address-model=64 cxxflags=-clr asynch-exceptions=on
     - APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2017
       ARGS: --toolset=msvc-14.1 address-model=32 cxxflags=-clr asynch-exceptions=on
+    - APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2017
+      ARGS: --toolset=msvc-14.1 architecture=arm testing.launcher=../../type_traits/test/null_runner.bat
```
Comment (janisozaur) [inline appveyor.yml]: Wouldn't `testing.execute=off` suffice here?

File: include/boost/type_traits/detail/is_function_cxx_11.hpp [hunk 2]
```diff
@@ -96,7 +96,7 @@ namespace boost {
    template <class Ret, class ...Args BOOST_TT_NOEXCEPT_PARAM>
    struct is_function<Ret(Args..., ...)const volatile && BOOST_TT_NOEXCEPT_DECL> : public true_type {};
 
-#ifdef _MSC_VER
+#if defined(_MSC_VER) && !defined(_M_ARM) && !defined(_M_ARM64)
```
Comment (janisozaur) [inline include/boost/type_traits/detail/is_function_cxx_11.hpp]: In a purely hypothetical universe where MSVC supports _even more_ architectures, wouldn't that re-surface? Or is that too hypothetical?

**Comment by janisozaur** [comment]
https://github.com/boostorg/type_traits/blob/c65f41d4c01383bb650faa2e89a8a2e53bc18c46/appveyor.yml#L85 seems to be an executable that prints out options used to build it. That won't run on x86 when compiled for arm and will fail the build.

**Comment by janisozaur** [comment]
Thanks for such a prompt fix.

File: appveyor.yml (hunk 1 above)
Comment (jzmaddock) [reply appveyor.yml]: I didn't know that existed - but yes just what we need!

File: include/boost/type_traits/detail/is_function_cxx_11.hpp (hunk 2 above)
Comment (jzmaddock) [reply include/boost/type_traits/detail/is_function_cxx_11.hpp]: For now yes ;)  Besides the hypothetical new architecture may need the extra specialisations just as likely as not need them?

More likely IMO is that __vectorcall stops being the same as the default and becomes something different again on arm64 as it is on x64.  Or not.

**Comment by jzmaddock** [comment]
Closing - I've condensed the commits and merged to develop manually.

**Pull request closed without merging**